app/assets/javascripts/admin/order_cycles/controllers/incoming_controller.js.coffee

```diff
@@ -2,7 +2,7 @@ angular.module('admin.orderCycles').controller 'AdminOrderCycleIncomingCtrl', ($
   $controller('AdminOrderCycleExchangesCtrl', {$scope: $scope, ocInstance: ocInstance, $location: $location})
 
   $scope.view = 'incoming'
-
+  # NB: weirdly at this next line $scope.order_cycle.id comes out undefined so we use $scope.order_cycle_id instead
```

Comment:
It looks like a race condition, `$scope.order_cycle` is loaded in `AdminEditOrderCycleCtrl` which this controller depend on via `AdminOrderCycleExchangesCtrl`. So if you try to access `$scope.order_cycle` before it's loaded, you'll get `undefined`. From memory, you could get around that by watching for changes on `$scope.order_cycle` via `$watch` before using it.

Reply:
i guessed as much. would a watch be preferable here, what do you think?

Reply:
I wouldn't bother given that you can access the `order_cycle_id` directly and we don't need any other order cycle data